src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -14,71 +14,85 @@
 
     private ReadFilterLibrary(){ /*no instance*/ }
 
-    /**
-     * local classes for static read filters
-     */
+    /** Do not filter out any read. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class AllowAllReadsReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){return true;}}
 
+    /** Filter out reads containing skipped region from the reference (CIGAR strings with 'N' operator). */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     //Note: do not call getCigar to avoid creation of new Cigar objects
     public static class CigarContainsNoNOperator extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! CigarUtils.containsNOperator(read.getCigarElements());}}
 
+    /** Keep only reads that are first of pair. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class FirstOfPairReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test (final GATKRead read) {
             return read.isFirstOfPair();}}
 
+    /**
+     * Keep only reads containing good CIGAR strings:
+     * <ul>
+     *     <li>Valid regarding the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications.</a></li>
+     *     <li>Does not start or end with deletions (with or without preceding clips) </li>
+     *     <li>Has not consecutive deletions/insertions operators.</li>
```

Comment:
Oh right -- we should definitely put that summary line in the annotation, and have the javadoc reserved for details the same way we do for other documented features.